docs/DeveloperGuide.md

```diff
@@ -209,40 +213,97 @@ Similarly, how an undo operation goes through the `Model` component is shown bel
 
 #### Design considerations:
 
 **Aspect: How undo & redo executes:**
 
-* **Alternative 1 (current choice):** Saves the entire address book.
-  * Pros: Easy to implement.
-  * Cons: May have performance issues in terms of memory usage.
-
-* **Alternative 2:** Individual command knows how to undo/redo by
-  itself.
-  * Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-  * Cons: We must ensure that the implementation of each individual command are correct.
-
-_{more aspects and alternatives to be added}_
+* **Alternative 1 (selected choice):** Save the entire address book as objects.
+    * Pros:
+      * Easy to implement.
+      * Unlikely to have bugs.
+    * Cons:
+      * May have performance issues in terms of memory usage.
+</li>
+
+* **Alternative 2:** Save the entire address book into storage as JSON files.
+    * Pros:
+      * Saves history between different program launches.
+    * Cons:
+      * May have performance issues with many storage accesses.
+      * Increases coupling as `Model` will now need a reference to `Storage`.
+</li>
+
+* **Alternative 3:** Save the entire address book as a JSON `String`.
+    * Pros:
+      * Reduces memory footprint as compared to objects.
+      * Faster than JSON files as there are no accesses to storage.
+    * Cons:
+      * May have performance issues as it has to be deserialized each time.
+</li>
+
+* **Alternative 4:** Individual command knows how to undo/redo by itself.
+    * Pros:
+      * Will use less memory (e.g. for `delete`, just save the person being deleted).
+      * Best performance.
+    * Cons:
+      * We must ensure that the implementation of each individual `undo` command is correct.
+        This would be especially difficult for commands that modify multiple people at once (e.g. `asset` editing commands)
+</li>
+
+**Aspect: Data structure used to store undo & redo states:**
+
+* **Alternative 1 (selected choice):** 2 Stacks.
+    * Pros:
+      * Easy to implement.
+      * Simple data structure.
+      * Easy to clear all redo states.
+    * Cons:
+      * May have performance issues if many redo states are cleared at once.
+</li>
+
+* **Alternative 2:** ArrayList, using pointers for current state and redo limit.
+    * Pros:
+      * Redo states no longer have to be cleared, as they are tracked by pointers and can be replaced at indexes as needed.
+    * Cons:
+      * Harder to implement.
+      * `add()` and `set()` have to be used appropriately to prevent synchronization issues.
+      * Pointers have to be carefully implemented.
+</li>
+
+* **Alternative 3:** LinkedList.
+    * Pros:
+      * Fast in dropping many nodes after a specified index.
+      * Simple data structure.
+    * Cons:
+      * Time-consuming to implement: Unfortunately, the built-in LinkedList does not have a method to drop all nodes after a certain index,
+        and hence a custom data structure would have to be used.
```

Comment:
Why would a custom data structure be necessary? The developer can call `linkedList.removeLast()` until the required size.

Reply:
The reasoning to use a linked list instead was to possibly make removing all nodes after a certain index O(1).
Using linkedList.removeLast() multiple times would make it O(N), resulting in no benefits of using a linked list as compared to other data structures.
I will update the description in the DG